client/src/components/DisplayTable/DisplayTableUtils.js

```diff
@@ -157,8 +157,17 @@ export const PageSelector = ({
   );
 };
 
-export const PageinateBySelector = ({ selected, on_select, num_items }) => {
-  const options = _.filter([100, num_items > 100 && num_items]);
+export const SelectPageSize = ({
+  selected,
+  on_select,
+  page_size_increment,
+  num_items,
+}) => {
+  const options = _.chain(num_items / page_size_increment)
+    .ceil()
+    .range()
+    .map((_, ix) => (ix + 1) * page_size_increment)
+    .value();
```

Comment:
Could be better, but exactly how was an unanswered design question in the base PR. Do we only want a fixed number of options? An explicit "show all" button?

Reply:
IMO, 100 - 500 then a show al button would be good. Right now the program expenditures by authority type table (RPB) has 36 page increment sizes which is making the dropdown gigantic haha.

Thoughts?

Reply:
Haha, yeah, that's too many options. Sounds good to me, we don't really want multiple large options since switching between those will be slow anyway, maybe only give a couple options less than something reasonable (whatever number of cells most tables start getting chunky at) and then the "all" button.

Something I forgot to leave a comment on but was an intentional part of my changes here. In the base branch, 100 was the hardcoded page size in the `DisplayTable` itself, where it only showed up in one place and, thus, would look like it was trivial to change later. The wrinkle to that was that, in a separate module, the page size selection component was also hardcoded to work in units of 100's. Generally some sort of configuration option like page size should only have a single source of truth (or better yet, an overridable default) at the top layer of the abstraction (the `DisplayTable`), and any internal components (like the utils) should know nothing/make no assumptions about it on their own. 

Another way to put it is that you should try to keep code (and concepts represented inside that code) "perpendicular", meaning only intersecting at a single point, whenever possible. Code-wise, it's easy if you're working with something like a React component, where everything has a clear, single interface and is, by design, composable (... although CSS can create a second, messy, layer over all this). A change in either should, **at most**, require a change at the interface of the other. The harder level is keeping track of concepts/assumptions/knowledge. In this example, they were coupled both by their interface and by their (indirect) shared knowledge of the magic number 100.

Assumptions, for example, tangle things up if they aren't explicitly checked at the interface. This is heavily formalized in strongly typed languages, which generally also take care of checking the assumptions and reminding you of them prior to run-time. Of course, you still need to architect your code and put in the work to define these assumptions in the first place, but at least future devs have more guide rails in place when you've done that.

... a THIRD way you might hear basically the same idea communicated is when someone talks about a "leaky abstraction". Organizing code in modules/methods/components/objects etc is as much about abstraction as it is about reusability. As you probably know, abstraction means representing a more complex concept with a different, but workably equivalent, simple concept. When talking about code organization, that mostly means hiding the implementation details behind a name and interface. An abstraction becomes leaky if it makes (undeclared or unsafe) assumptions about where and how it will be used (generally, this is when we get a bug time bomb), or if too many implementation details surface up through the interface (generally, this is when the abstraction becomes useless as developers need to learn the implementation details anyway; may also couple the abstraction too tightly to the consumers and become brittle/hard to refactor in the future).